darts/models/forecasting/prophet_model.py

```diff
@@ -88,6 +91,10 @@ def __init__(
         self.country_holidays = country_holidays
         self.prophet_kwargs = prophet_kwargs
         self.model = None
+        self.suppress_stdout_stderr = suppress_stdout_stderror
+
+        self.execute_and_suppress_output = execute_and_suppress_output
+        self.model_builder = prophet.Prophet
```

Comment:
Why is this needed? It was somewhat simpler to call `prophet.Prophet` directly in the `fit()` method, no?

Reply:
This is needed for unit testing, because I think we don't need to invoke the actual fit to test this feature; we only want to make sure that method fit is called. Secondly, in the previous approach, with Prophet model being constructed for every fit, I cannot really check if fit is called.

Reply:
i.e. if I mock this `model_builder`

```
model = darts.Prophet(...)
model.model_builder = Mock(return_value=Mock(fit=Mock(return_value=True)))
```

Then I can:
1. Avoid calling the real fit
2. Check if fit is called once

Reply:
I think another unit test is needed to make sure that the default `model_builder = prophet.Prophet`

Reply:
OK, got it, makes sense indeed, I missed this in the unit tests. Yes, maybe the extra unit test you suggest would be nice too.

Reply:
OK, got it, makes sense indeed, I missed this in the unit tests. Yes, maybe the extra unit test you suggest would be nice too.